AsyncUsageAnalyzers/AsyncUsageAnalyzers/Naming/UseAsyncSuffixAnalyzer.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace AsyncUsageAnalyzers.Naming
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    /// <summary>
+    /// This analyzer identifies asynchronous methods using the Task-based Asynchronous Pattern (TAP) according to their
+    /// signature, and reports a warning if the method name does not include the suffix <c>Async</c>.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class UseAsyncSuffixAnalyzer : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="UseAsyncSuffixAnalyzer"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "UseAsyncSuffix";
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(NamingResources.UseAsyncSuffixTitle), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(NamingResources.UseAsyncSuffixMessageFormat), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly string Category = "AsyncUsage.CSharp.Naming";
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(NamingResources.UseAsyncSuffixDescription), NamingResources.ResourceManager, typeof(NamingResources));
+        private static readonly string HelpLink = null;
```

Comment:
You could write a help page on the wiki part of this project and link to it here.